Ug changes from pe

Gabriel4357 opened the pull request.

Gabriel4357 committed:
Merge branch 'master' of https://github.com/AY2324S1-CS2103T-T13-3/tp

* 'master' of https://github.com/AY2324S1-CS2103T-T13-3/tp:
  Update UserGuide.md
  Fix some upper case issues

Gabriel4357 committed:
Fix a few bugs pointed out from PE

These include adding a Register portion above our
Getting Started portion to avoid confusion.

Also, this includes changing the use of a longer - to a proper
- so that examples can be copied and paste into the
Command Box.

Gabriel4357 committed:
Change certain layout and make changes to UG

These includes :
*changing order to TOC
*change all Customer and Delivery to upper case C and D
* refine the order or certain messages
* adds FAQ about corrupted data

B-enguin commented on docs/UserGuide.md:
```diff
@@ -23,6 +23,52 @@ Essentially, with HomeBoss, you can **focus on what matters most: running your b
+# Table of Contents
+
+* 1\. [About HomeBoss](#about-homeboss)
+* 2\. Table of Contents
```
Are we going to hyperlink this?

B-enguin commented on docs/UserGuide.md:
```diff
@@ -268,10 +267,10 @@ logout
 * Words that are in `UPPER_CASE`{.swift} are parameters that are to be supplied by you.
 * Parameters that are preceded by a prefix must be supplied after that prefix, e.g. `--name DELIVERY_NAME`{.swift}
-  must be given as `--name deliveryName`{.swift} and not `deliveryName`{.swift} or `--name`{.swift}
+  must be given as `--name DELIVERY_NAME`{.swift} and not `DELIVERY_NAME`{.swift} or `--name`{.swift}
```
I think this is supposed to be an example of the parameter not the parameter itself

Gabriel4357 replied:
right ok changed to actual example

B-enguin commented on docs/UserGuide.md:
```diff
@@ -268,10 +267,10 @@ logout
 * Items that are placed in square brackets (`[]`{.swift}) are optional, e.g. `DELIVERY_ID [--name DELIVERY_NAME]`
   {.swift} can either be
-  `1`{.swift} or `1 --name deliveryName`{.swift}.
+  `1`{.swift} or `1 --name DELIVERY_NAME`{.swift}.
```
Same as above

Gabriel4357 replied:
yep ok

B-enguin commented on docs/UserGuide.md:
```diff
@@ -284,13 +283,14 @@ logout
 
 1. First register for HomeBoss using the `register`{.swift} command, more details on the command can be found
-   [here](#register)
+   [here](#register). So, for example, type `register --user Gabriel --password GabrielIsGreat --confirmPass 
+   GabrielIsGreat --secretQn First Pet Name? --answer Koko`{.swift} into the Command Box and hit enter.
```
Should the more details be placed at the end rather than the middle?

Gabriel4357 replied:
sure

B-enguin commented on docs/UserGuide.md:
```diff
@@ -284,13 +283,14 @@ logout
 
 1. First register for HomeBoss using the `register`{.swift} command, more details on the command can be found
-   [here](#register)
+   [here](#register). So, for example, type `register --user Gabriel --password GabrielIsGreat --confirmPass 
+   GabrielIsGreat --secretQn First Pet Name? --answer Koko`{.swift} into the Command Box and hit enter.
```
Also Would the register command benefit from a format like the add command below, where we specifically state what are the parameters involved and their example value, e.g. USERNAME: HomeBoss, PASSWORD: Password

Gabriel4357 replied:
yup good idea done

B-enguin commented on docs/UserGuide.md:
```diff
@@ -300,33 +300,35 @@ by adding your first Delivery
-3. Let's assume that a new customer, Gabriel, is to be added, using the `customer add`{.swift} command.
 
-> `NAME`: Gabriel
+3. Let's assume that a new Customer, Alex, is to be added, using the `customer add`{.swift} command.
+
+> `NAME`: Alex
```
Would it be better to keep the name as Gabriel as the values were specifically chosen to match the screenshot

Gabriel4357 replied:
yep didnt realise thx

B-enguin commented on docs/UserGuide.md:
```diff
@@ -300,33 +300,35 @@ by adding your first Delivery
+6. Now, let's add a new Delivery for Alex into HomeBoss, suppose Alex ordered some furniture.
 
-> `DELIVERY_NAME`: Macbook Pro
+> `DELIVERY_NAME`: furniture
 > `DELIVERY_DATE`: 2023-12-03
```
Same as above, the values were specifically chosen because of the screenshot

Gabriel4357 replied:
kk

B-enguin commented on docs/UserGuide.md:
```diff
@@ -1063,6 +1065,16 @@ A: It is currently not possible to export your data to another software.
+data to be invalid? **
+
+A: In the rare event that this occurs, if the Customer data is the one that is edited and becomes invalid, all the 
+Customer data will be wiped completely and even the unique IDs of the Customers will start from 1. (even a clear 
+command will not cause the unique IDs to start from 1) The Delivery data,however, will still be intact.
```
Do we need to mention the clear command

B-enguin replied:
I believe the Delivery Data would get wiped if the Customer Data is invalid since delivery references customer

B-enguin commented on docs/UserGuide.md:
```diff
@@ -1078,27 +1090,27 @@ A: HomeBoss only supports English characters.
+| View    | `customer view CUSTOMER_ID`                                                                          | `customer view 1`                                                                                     |
+| List    | `customer list`                                                                                      | `customer list`                                                                                       |
+| Find    | `customer find KEYWORD [MORE_KEYWORDS]`                                                              | `customer find Ibrahim Yu`                                                                            |
+| Edit    | `customer edit CUSTOMER_ID [--name NAME] [--phone PHONE_NUMBER] [--email EMAIL] [--address ADDRESS]` | `customer edit 1 --name Gabriel -–phone 1234 5678 --email gabrielrocks@gmail.com --address RVRC Block B` |
```
Example for phone prefix, I think the dashes used are different

Gabriel4357 replied:
done

B-enguin reviewed (commented):
Some general comments

Gabriel4357 committed:
Update based off comments

Make a few changes including:
* refining and adding more details to Getting Started
*fix some minor errors
*change phrasing of the FAQ portion

juliusgambe commented on docs/UserGuide.md:
```diff
@@ -284,13 +283,22 @@ logout
+1. First register for HomeBoss using the `register`{.swift} command. So, for example, if you want to register an account
+   with the following details:
+
+> `USERNAME`: Alex123
+> `PASSWORD`: AlexIsGreat
+> `CONFIRM_PASSWORD`: AlexIsGreat
+> `SECRET_QUESTION`: First Pet Name? 
+> `ANSWER`: KoKo 
```
Could you help me include `{.swift}` after the code block? Thank you : 3

Gabriel4357 replied:
doneee

juliusgambe commented on docs/UserGuide.md:
```diff
@@ -669,52 +679,52 @@ worries! You can edit the details of a customer simply by calling this command.
 ### Add a Delivery
 
-> Adds a delivery to the delivery book.
+> Adds a Delivery to the delivery book.
```
Actually, we're removing this line and merge them together with the description below right?

Gabriel4357 replied:
yep

Gabriel4357 committed:
Merge branch 'master' into UG-changes-fromPE

juliusgambe reviewed (approved):
LGTM
